### DISCUSSION TITLE
Make samples use Metal instead of OpenGL on macOS?

### DISCUSSION THREAD

**Discussion by jwwalker** [google/filament#3627] [Q&A] [answered]
How can I build the samples to use Metal on macOS?  I am not very familiar with CMake and Ninja.  A commit on March 2 [1189c9e29] seriously broke my Metal rendering, and if I could reproduce the problem in a sample, that would be easier than trying to make one of my Xcode projects self-contained enough to put in a bug report.

**Comment by romainguy** [comment, marked as answer]
There's nothing special to do. The main thing is to request a Metal backend when you create the Engine in your app. Some of our samples like gltf_viewer let you pick the backend with a command line flag (`-a metal` when running the sample).

**Reply by romainguy** [reply]
Also @pixelflinger who wrote that big refactor

**Reply by jwwalker** [reply]
Thanks.  For the record, `rendertarget` is another sample that accepts `-a metal`.

**Reply by jwwalker** [reply]
For posterity I will also note that one can change the backend for all the samples that don't take command line parameters by editing `filament/libs/filamentapp/include/filamentapp/Config.h`, where it sets the default value of `Config.backend`.

**Reply by romainguy** [reply]
You can do a lot of things by changing the source code ^^

**Reply by jwwalker** [reply]
My point was that one doesn't need to change the source code of each sample individually.
